src/access.rs

```diff
@@ -0,0 +1,438 @@
+		let mut unconfirmed_registered_txs = HashSet::new();
+
+		for txid in registered_txs {
+			if let Some(confirmed_tx) = self.get_confirmed_tx(&txid, None, None).await? {
+				confirmed_txs.push(confirmed_tx);
+			} else {
+				unconfirmed_registered_txs.insert(txid);
+			}
+		}
+
+		// Check all registered outputs for dependent spending transactions.
+		let registered_outputs = self.watched_outputs.lock().unwrap().clone();
+
+		// Remember all registered outputs that haven't been spent for future processing.
+		let mut unspent_registered_outputs = HashSet::new();
+
+		for output in registered_outputs {
+			if let Some(output_status) = client
+				.get_output_status(&output.outpoint.txid, output.outpoint.index as u64)
+				.await?
+			{
+				if let Some(spending_txid) = output_status.txid {
+					if let Some(spending_tx_status) = output_status.status {
+						if let Some(confirmed_tx) = self
+							.get_confirmed_tx(
+								&spending_txid,
+								spending_tx_status.block_hash,
+								spending_tx_status.block_height,
+							)
+							.await?
+						{
+							confirmed_txs.push(confirmed_tx);
+							continue;
+						}
+					}
+				}
+			}
+			unspent_registered_outputs.insert(output);
+		}
+
+		// Sort all confirmed transactions first by block height, then by in-block
+		// position, and finally feed them to the interface in order.
+		confirmed_txs.sort_unstable_by(|tx1, tx2| {
+			tx1.block_height.cmp(&tx2.block_height).then_with(|| tx1.pos.cmp(&tx2.pos))
+		});
+
+		Ok((confirmed_txs, unconfirmed_registered_txs, unspent_registered_outputs))
+	}
+
+	async fn get_confirmed_tx(
+		&self, txid: &Txid, expected_block_hash: Option<BlockHash>, known_block_height: Option<u32>,
+	) -> Result<Option<ConfirmedTx>, Error> {
+		let client = &*self.blockchain;
+		if let Some(merkle_block) = client.get_merkle_block(&txid).await? {
+			let block_header = merkle_block.header;
+			let block_hash = block_header.block_hash();
+			if let Some(expected_block_hash) = expected_block_hash {
+				if expected_block_hash != block_hash {
+					return Err(Error::ChainAccessInconsistency);
+				}
+			}
+
+			let mut matches = vec![*txid];
+			let mut indexes = Vec::new();
+			let _ = merkle_block.txn.extract_matches(&mut matches, &mut indexes);
+			assert_eq!(indexes.len(), 1);
+			let pos = indexes[0] as usize;
+			if let Some(tx) = client.get_tx(&txid).await? {
+				if let Some(block_height) = known_block_height {
+					// We can take a shortcut here if a previous call already gave us the height.
+					return Ok(Some(ConfirmedTx { tx, block_header, pos, block_height }));
+				}
+
+				let block_status = client.get_block_status(&block_hash).await?;
+				if let Some(block_height) = block_status.height {
+					return Ok(Some(ConfirmedTx { tx, block_header, pos, block_height }));
+				} else {
+					// If any previously-confirmed block suddenly is no longer confirmed, we found
+					// an inconsisteny and should start over.
```

Comment:
nit typo: > inconsisteny
inconsistency

Reply:
Thanks! Fixed in https://github.com/lightningdevkit/ldk-lite/pull/9/commits/d9bc655e249fedc46f2c6f510fcc9374344d8ea1.